src/logger.rs

```diff
@@ -1,34 +1,30 @@
 use crate::serial::Serial;
-use core::{
-    fmt::write,
-    format_args,
-    sync::atomic::{AtomicU16, Ordering},
-};
+use core::{fmt::write, format_args};
 use log::*;
+use spin::Mutex;
 
 const COM1_PORT: u16 = 0x3f8;
 
-static LOGGER: Logger = Logger(AtomicU16::new(COM1_PORT));
+static LOGGER: Logger = Logger(Mutex::new(COM1_PORT));
 
-struct Logger(AtomicU16);
+struct Logger(Mutex<u16>);
 
 impl log::Log for Logger {
     fn enabled(&self, _m: &Metadata) -> bool {
         true
     }
 
     fn log(&self, record: &Record) {
-        let mut serial = Serial::new(self.0.load(Ordering::Relaxed));
+        let port = self.0.lock();
+        let mut serial = Serial::new(*port);
 
         let _ = write(
             &mut serial,
             format_args!(
-                "{:>8}: {} ({}, {}:{})\n",
+                "[{}  {}] {}\n\0",
                 record.level(),
-                record.args(),
-                record.target(),
-                record.file().unwrap_or("<unknown>"),
-                record.line().unwrap_or(0),
+                record.module_path().unwrap_or(record.target()),
+                record.args()
```

Comment:
Why is this format change required?

Reply:
no its not required, it can be discarded

Reply:
Can you discard the format change part? The other part LGTM 👍

Reply:
@Azvanzed Can you discard the format change part? The other part LGTM 👍

Reply:
Let me handle this :+1: